crates/cashu/src/nuts/nut11.rs

```diff
@@ -237,7 +236,11 @@ impl TryFrom<P2PKConditions> for Secret {
             tags.push(Tag::Refund(refund_keys).as_vec())
         }
 
-        tags.push(Tag::SigFlag(sig_flag).as_vec());
+        if !sig_flag.is_default() {
+            tags.push(Tag::SigFlag(sig_flag).as_vec());
+        }
+
+        let tags = if !tags.is_empty() { Some(tags) } else { None };
```

Comment:
you can use `then` to avoid the if/else:

```suggestion
        let tags = (!tags.is_empty()).then(|| tags);
```